state/state_test.go

```diff
@@ -121,6 +164,60 @@ func TestStateSaveLoad(t *testing.T) {
 	assert.True(state.Equals(loadedState),
 		fmt.Sprintf("expected state and its copy to be identical.\ngot: %v\nexpected: %v\n",
 			loadedState, state))
+
+	// the following assertions are just for additional assurance
+	assert.Equal(state.TimeoutCommit, loadedState.TimeoutCommit,
+		fmt.Sprintf("expected TimeoutCommit to be equal."+
+			"\ngot: %v\nexpected: %v\n", loadedState.TimeoutCommit, state.TimeoutCommit))
+	assert.Equal(state.TimeoutPropose, loadedState.TimeoutPropose, fmt.Sprintf("expected TimeoutPropose to be equal."+
+		"\ngot: %v\nexpected: %v\n", loadedState.TimeoutPropose, state.TimeoutPropose))
+}
+
+// TestLoadConsensusTimeoutsInfo tests whether timeouts are properly saved
+// while saving State to a db and can be retrieved using LoadConsensusTimeoutsInfo.
+func TestLoadConsensusTimeoutsInfo(t *testing.T) {
+	tearDown, stateDB, state := setupTestCase(t)
+	defer tearDown(t)
+	stateStore := sm.NewStore(stateDB, sm.StoreOptions{
+		DiscardABCIResponses: false,
+	})
+	assert := assert.New(t)
+
+	// add two states to the store
+	state1 := state.Copy()
+	state1.TimeoutCommit = 1 * time.Second
+	state1.TimeoutPropose = 2 * time.Second
+	// advance height to ensure timeouts are saved for a new height
+	state1.LastBlockHeight++
+	state1.LastValidators = state.Validators
+	err := stateStore.Save(state1)
+	require.NoError(t, err)
+
+	state2 := state.Copy()
+	// advance height by 2 to ensure timeouts are
+	// saved for a new height
+	state2.LastBlockHeight += 2
+	state2.TimeoutCommit = 3 * time.Second
+	state2.TimeoutPropose = 4 * time.Second
+	state2.LastValidators = state.Validators
+	err = stateStore.Save(state2)
+	require.NoError(t, err)
+
+	// now try to retrieve saved sates from the store, focusing on timeouts
```

Comment:
```suggestion
	// now try to retrieve saved states from the store, focusing on timeouts
```

Reply:
resolving this as we removed this test entirely since we are no longer saving or loading timeouts from the state